### PR TITLE
Added Chroot SimProcedures

### PR DESCRIPTION
**Pull request by Aydalanche**

### CODE REVIEW

File: angr/procedures/posix/chroot.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+            if newPath[size:size+1] == "\\":
+                newPath = newPath[2:size]
+                break
+            size += 1
+
+        if not os.path.exists(newPath):
+            return -1
+        
+        self.state.fs.mount('/', angr.SimHostFilesystem(newPath))
+        return 0
```
Comment (ltfish) [inline angr/procedures/posix/chroot.py:20]: Please add a blank line at the end of the file.

File: tests/test_chroot.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    #Try and read the files stat size from new chrooted dir
+    try:
+        fd = simgr.deadended[0].posix.open('/test.txt', Flags.O_RDONLY)
+        stat = simgr.deadended[0].posix.fstat(fd)
+        print(f"File's Stat Size: {stat.st_size}")
+    except:
+        print("Error while trying to open the file")
+
+if __name__ == "__main__":
+    test_chroot()
```
Comment (ltfish) [inline tests/test_chroot.py:24]: Same here. A blank line at the end of the file.

File: tests/test_chroot.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import angr
+from angr.state_plugins.posix import Flags
+
+import os
+test_location = os.path.join(os.path.dirname(os.path.realpath(__file__)), '..', '..', 'binaries', 'tests')
+
+def test_chroot():
```
Comment (ltfish) [inline tests/test_chroot.py:7]: Please switch to `unittest.TestCase`-style test cases.

File: tests/test_chroot.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+    simgr.run()
+
+    #Try and read the files stat size from new chrooted dir
+    try:
+        fd = simgr.deadended[0].posix.open('/test.txt', Flags.O_RDONLY)
+        stat = simgr.deadended[0].posix.fstat(fd)
+        print(f"File's Stat Size: {stat.st_size}")
+    except:
+        print("Error while trying to open the file")
```
Comment (ltfish) [inline tests/test_chroot.py:21]: A test case should fail by raising exceptions or assertion failures when success conditions are not met.

**Review by ltfish — CHANGES_REQUESTED** [review]

**Comment by Aydalanche** [comment]
I updated the code to add what you requested and I removed the try and except part of the code from the test file so that it will actually fail if the file or filesystem can't be found.

**Review by ltfish — APPROVED** [review]

**Comment by ltfish** [comment]
Merged. Thanks!

**Pull request merged**